textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/TranslatorUtils.java

```diff
@@ -0,0 +1,150 @@
+package edu.uci.ics.textdb.dataflow.regexmatch;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * @author Zuozhi Wang
+ * @author Shuying Lai
+ *
+ */
+
+public class TranslatorUtils {
+
+	static final int MAX_EXACT_SIZE = 7;
+	
+	/** Prefix and suffix sets are limited to maxSet strings.
+	 * If they get too big, simplify will replace groups of strings
+	 * sharing a common leading prefix (or trailing suffix) with
+	 * that common prefix (or suffix).  It is useful for maxSet
+	 * to be at least 2³ = 8 so that we can exactly
+	 * represent a case-insensitive abc by the set
+	 * {abc, abC, aBc, aBC, Abc, AbC, ABc, ABC}.
+	 */
+	static final int MAX_SET_SIZE = 20;
+	
+	/**
+	 * This function interfaces provide a method to fold
+	 * (concat / alternate) two {@code RegexInfo} objects.
```

Comment:
I am not following this... The fold function is supposed to receive two `RegexInfo` and return one `RegexInfo`. Should we say "fold two into one"? I am not familiar with usage of the word "fold" though.